### PR TITLE
syzkaller: update go get invocation

### PR DESCRIPTION
**Pull request by dvyukov**
The way to checkout the repo has changed.
Fixes: https://bugs.chromium.org/p/oss-fuzz/issues/detail?id=21994

### CODE REVIEW

**Comment by dvyukov** [comment]
I gave up. Where are my sourced, dude?
We need to run 'make descriptions' in the syzkaller dir before building fuzzers, but I can't comprehend how this can be done.

File: projects/syzkaller/Dockerfile [hunk 1]
```diff
@@ -17,11 +17,11 @@
 FROM gcr.io/oss-fuzz-base/base-builder
 MAINTAINER mmoroz@chromium.org
 
-RUN go get -u -d github.com/google/syzkaller/...
+RUN go get -u -d github.com/google/syzkaller/prog
```
Comment (Dor1s) [inline projects/syzkaller/Dockerfile:20]: sorry I can't debug myself (see go/mmoroz-ooo), but my first guess would be that this line isn't doing what we expect it to do

you can also try this locally, check out OSS-Fuzz repo and follow https://google.github.io/oss-fuzz/getting-started/new-project-guide/#prerequisites, then do:

```
python infra/helper.py build_image syzkaller
python infra/helper.py shell syzkaller
```

that would open bash inside syzkaller build image, i.e. you can explore the file system interactively and/or try checking out the sources some other way

**Comment by dvyukov** [comment]
This helped to "fix" it:
```
python infra/helper.py build_image syzkaller
python infra/helper.py shell syzkaller
```

Now the build is broken with:
```
+ go-fuzz -func Fuzz -o compiler_fuzzer.a ./pkg/compiler
unexpected directory layout:
	import path: _/root/go/src/github.com/google/syzkaller/pkg/compiler
	root: /root/go/src
	dir: /root/go/src/github.com/google/syzkaller/pkg/compiler
	expand root: /root/go
	expand dir: /root/go/src/github.com/google/syzkaller/pkg/compiler
	separator: /
```

But that's a separate problem. @mdempsky, this go-fuzz is yours as far as I remember, right?

**Comment by inferno-chromium** [comment]
> This helped to "fix" it:
> 
> ```
> python infra/helper.py build_image syzkaller
> python infra/helper.py shell syzkaller
> ```
> 
> Now the build is broken with:
> 
> ```
> + go-fuzz -func Fuzz -o compiler_fuzzer.a ./pkg/compiler
> unexpected directory layout:
> 	import path: _/root/go/src/github.com/google/syzkaller/pkg/compiler
> 	root: /root/go/src
> 	dir: /root/go/src/github.com/google/syzkaller/pkg/compiler
> 	expand root: /root/go
> 	expand dir: /root/go/src/github.com/google/syzkaller/pkg/compiler
> 	separator: /
> ```
> 
> But that's a separate problem. @mdempsky, this go-fuzz is yours as far as I remember, right?

Yes right, we are using @mdempsky's go-fuzz everywhere, can you please take a look.

**Comment by Dor1s** [comment]
Is this obsolete or still needs review?

**Comment by inferno-chromium** [comment]
syzkaller build is still broken and we didnt get response from @mdempsky :(

**Comment by Dor1s** [comment]
> syzkaller build is still broken and we didnt get response from mdempsky :(

I've made a change that should fix this, let's wait for CI to confirm.

**Pull request merged**